src/app/pages/api-keys/store/api-key.store.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { EMPTY } from 'rxjs';
+import {
+  catchError, tap,
+} from 'rxjs/operators';
+import { ApiKey } from 'app/interfaces/api-key.interface';
+import { EntityUtils } from 'app/modules/entity/utils';
+import { DialogService, WebSocketService } from 'app/services';
+
+export interface ApiKeysState {
+  isLoading: boolean;
+  error: string;
+  entities: ApiKey[];
+}
+
+const initialState: ApiKeysState = {
+  isLoading: false,
+  entities: [],
+  error: null,
+};
+
+@Injectable()
+export class ApiKeyComponentStore extends ComponentStore<ApiKeysState> {
+  constructor(
+    private ws: WebSocketService,
+    private dialog: DialogService,
+  ) {
+    super(initialState);
+  }
+
+  readonly apiKeys$ = this.select((state) => state.entities);
+  readonly apiKeysTotal$ = this.select((state) => state.entities.length);
+
+  readonly loadApiKeys = this.effect(() => {
+    this.setState({
+      ...initialState,
+      isLoading: true,
+    });
+
+    return this.ws.call('api_key.query').pipe(
+      tap((keys) => {
+        this.patchState({
+          isLoading: false,
+          entities: keys,
+        });
+      }),
+      catchError((error) => {
+        new EntityUtils().errorReport(error, this.dialog);
+
+        this.patchState({
+          isLoading: false,
+          error: 'API Keys could not be loaded',
+        });
+
+        return EMPTY;
+      }),
+    );
+  });
+
+  addApiKey = this.updater((state, key: ApiKey) => ({
```

Comment:
I think one of two things should happen. Either we should include websocket calls for add/edit/delete in store (probably better?) or we should rename these to `apiKeyAdded`, `apiKeyEdited`, etc.

Reply:
> I think one of two things should happen. Either we should include websocket calls for add/edit/delete in store (probably better?) or we should rename these to `apiKeyAdded`, `apiKeyEdited`, etc.

Since the first one is not working as expected. Probably make sense to go the second way.